### PR TITLE
docs: update to use PROXY config

### PR DESCRIPTION
**Pull request by kyessenov**
Documents https://github.com/istio/istio/pull/45987

### CODE REVIEW

File: content/en/docs/ops/common-problems/upgrade-issues/index.md [hunk 1]
```diff
@@ -102,14 +104,52 @@ spec:
...
-Use the equivalent ingress gateway proxy configuration annotation:
+Use the equivalent ingress gateway pod proxy configuration annotation:
 
 {{< text yaml >}}
 metadata:
   annotations:
     "proxy.istio.io/config": '{"gatewayTopology" : { "numTrustedProxies": 1 }}'
 {{< /text >}}
 
+### Use gateway topology to enable PROXY protocol on the ingress gateways
```
Comment (windsonsea) [inline content/en/docs/ops/common-problems/upgrade-issues/index.md:115]:
```suggestion
### Use gateway topology to enable proxy protocol on the ingress gateways
```
Checked `proxy protocol` in the entire repo. Is it necessary to capitalize all letters with `PROXY`?

image

File: content/en/docs/ops/common-problems/upgrade-issues/index.md [hunk 2]
```diff
@@ -102,14 +104,52 @@ spec:
...
+
+The usage of `EnvoyFilter` to enable [PROXY
+protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) on the
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply content/en/docs/ops/common-problems/upgrade-issues/index.md:118]: This is markdown and will be reflown? It is easier to read fixed width sources.

File: content/en/docs/ops/common-problems/upgrade-issues/index.md (hunk 1 above)
Comment (dhawton) [reply content/en/docs/ops/common-problems/upgrade-issues/index.md:115]: I agree when referencing the PROXY Protocol as defined, we should capitalize to distinguish it similar to the RFC.

File: content/en/docs/ops/common-problems/upgrade-issues/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (windsonsea) [reply content/en/docs/ops/common-problems/upgrade-issues/index.md:118]: If you search "PROXY protocol", this sentence will be ignored due to its broken by a newline char. Whatever, this is only a nit.

File: content/en/docs/ops/common-problems/upgrade-issues/index.md [hunk 3]
```diff
@@ -72,7 +72,9 @@ for deployment of Wasm code.
 ### Use gateway topology to set the number of the trusted hops
 
 The usage of `EnvoyFilter` to configure the number of the trusted hops in the
-HTTP connection manager has been replaced by the `gatewayTopology` field in
+HTTP connection manager has been replaced by the
+[`gatewayTopology`]((/docs/reference/config/istio.mesh.v1alpha1/#Topology)
```
Comment (dhawton) [inline content/en/docs/ops/common-problems/upgrade-issues/index.md:76]:
```suggestion
[`gatewayTopology`](/docs/reference/config/istio.mesh.v1alpha1/#Topology)
```

File: content/en/docs/ops/common-problems/upgrade-issues/index.md [hunk 4]
```diff
@@ -102,14 +104,52 @@ spec:
...
   annotations:
     "proxy.istio.io/config": '{"gatewayTopology" : { "numTrustedProxies": 1 }}'
 {{< /text >}}
 
+### Use gateway topology to enable PROXY protocol on the ingress gateways
+
+The usage of `EnvoyFilter` to enable [PROXY
+protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) on the
+ingress gateways has been replaced by the
+[`gatewayTopology`]((/docs/reference/config/istio.mesh.v1alpha1/#Topology)
```
Comment (dhawton) [inline content/en/docs/ops/common-problems/upgrade-issues/index.md:120]:
```suggestion
[`gatewayTopology`](/docs/reference/config/istio.mesh.v1alpha1/#Topology)
```

**Review by dhawton — CHANGES_REQUESTED** [review]

File: content/en/docs/ops/common-problems/upgrade-issues/index.md (hunk 3 above)
Comment (kyessenov) [reply content/en/docs/ops/common-problems/upgrade-issues/index.md:76]: Done

File: content/en/docs/ops/common-problems/upgrade-issues/index.md (hunk 4 above)
Comment (kyessenov) [reply content/en/docs/ops/common-problems/upgrade-issues/index.md:120]: Done

File: content/en/docs/ops/configuration/traffic-management/network-topologies/index.md [hunk 5]
```diff
@@ -246,60 +246,27 @@ PROXY protocol is only supported for TCP traffic forwarding by Envoy. See the [E
 PROXY protocol should not be used for L7 traffic, or for Istio gateways behind L7 load balancers.
 {{< /warning >}}
 
-If your external TCP load balancer is configured to forward TCP traffic and use the PROXY protocol, the Istio Gateway TCP listener must also be configured to accept the PROXY protocol. Enabling this requires adding the [Envoy Proxy Protocol filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/proxy_protocol) using an `EnvoyFilter` applied on the gateway workload. For example:
-
-{{< tabset category-name="config-api" >}}
-
-{{< tab name="Istio APIs" category-value="istio-apis" >}}
+If your external TCP load balancer is configured to forward TCP traffic and use the PROXY protocol, the Istio Gateway TCP listener must also be configured to accept the PROXY protocol.
+To enable PROXY protocol on all TCP listeners on the gateways, set `proxyProtocol` in your `IstioOperator`
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply content/en/docs/ops/configuration/traffic-management/network-topologies/index.md:250]: Done.

**Review by kyessenov — COMMENTED** [review]
Thank you for the review, PTAL.

**Comment by kyessenov** [comment]
@ericvn Can this move forward?

**Comment by ericvn** [comment]
@dhawton, would you review this PR again and remove the `request changes` if appropriate. Thanks.

**Comment by kyessenov** [comment]
@ericvn @linsun can we get this moving again?

**Review by dhawton — APPROVED** [review]

**Pull request merged**